test/src/unit-cppapi-float-scaling-filter.cc

```diff
@@ -73,22 +74,22 @@ TEST_CASE(
 template <typename T, typename W>
 struct FloatScalingFilterTestStruct {
   void float_scaling_filter_api_test(
-      Context& ctx, tiledb_array_type_t array_type) {
-    /// TODO: Comment whole function.
+      Context& ctx, tiledb_array_type_t array_type, bool negative) {
+    // Create the domain and dimensions.
     Domain domain(ctx);
     auto d1 = Dimension::create<int>(ctx, "rows", {{1, dim_hi}}, 4);
     auto d2 = Dimension::create<int>(ctx, "cols", {{1, dim_hi}}, 4);
     domain.add_dimensions(d1, d2);
 
+    // Create the filter and set the scale, offset,and byte width.
     Filter f(ctx, TILEDB_FILTER_SCALE_FLOAT);
-
     double scale = 2.53;
     double offset = 0.138;
```

Comment:
Why test with just these specific values?

Reply:
I tried playing with random values just now, and found out that the compiler optimizes the result out when the values are too large, causing a test failure that is not actually my float scaling filter codes fault, but the tests fault. I wish there was a way to compile tests with less than O3 but alas... 

I have submitted a change where the values are between -64 and 64, and this seems to pass. Perhaps we should investigate this further. I think I talked to Isaiah about this already and he just said to note that this is a lossy compressor...

Reply:
>the compiler optimizes the result out when the values are too large

That really doesn't sound right. Do you have specific values that were generated that we causing failures? (If not, it illustrates why using non-repeatable random numbers is bad for testing and using seedable, repeatable pseudorandom sequences is good for testing.)

"Too large" with scaling could mean that you're getting overflow, in which case the testing is correctly identifying a defect in the compressor, failing to recognize overflow and failing on compression, which is better than writing out bad data without a failure.

Reply:
I checked the original PR #3243. It has an overflow/underflow defect. There's no checking at all for either condition. The reason the error is not showing up with fixed test parameters is that the data is within the realm of validity for the filter parameters.

A simple example. With 56-bit mantissas (`double`), scale = 1 and offset = 2^11, then data `max_float - (2^10)` will have an addition overflow.

Reply:
is it better that I check for overflow/underflow with https://en.cppreference.com/w/cpp/numeric/fenv/feclearexcept or manually?